ext/spl/spl_directory.h

```diff
@@ -138,7 +138,7 @@ static inline spl_filesystem_object* spl_filesystem_iterator_to_object(spl_files
 #define SPL_FILE_DIR_KEY_AS_FILENAME       0x00000100 /* make RecursiveDirectoryTree::key() return getFilename() */
 #define SPL_FILE_DIR_FOLLOW_SYMLINKS       0x00000200 /* make RecursiveDirectoryTree::hasChildren() follow symlinks */
 #define SPL_FILE_DIR_KEY_MODE_MASK         0x00000F00 /* mask RecursiveDirectoryTree::key() */
-#define SPL_FILE_DIR_KEY(intern,mode)      ((intern->flags&SPL_FILE_DIR_KEY_MODE_MASK)==mode)
+#define SPL_FILE_DIR_KEY(intern,mode)      ((intern->flags&SPL_FILE_DIR_KEY_AS_FILENAME)==mode)
```

Comment:
I'd suggest to move SPL_FILE_DIR_FOLLOW_SYMLINKS down into the SPL_FILE_DIR_OTHERS_MASK category instead. As written, this macro doesn't really make sense.

Reply:
As in changing the value? I wasn't sure if that would be ok to do, unless it's fine since it's non BC anyway?

Currently SPL_FILE_DIR_KEY_MODE_MASK doesn't make sense, so changing the value would allow it to make sense again...

I can move it down too.

Reply:
At least for master, it's okay to change the value.